source/base/function_parser.cc

```diff
@@ -176,6 +179,27 @@ namespace internal
   {
     return erfc(value);
   }
+
+  // returns a random value in the range [0,1]
+  double mu_rand_seed(double seed)
+  {
+    std::mutex mtx;
+    mtx.lock();
+    static boost::random::mt19937 rng((unsigned int)seed);
+    mtx.unlock();
```

Comment:
You'll have to make the `mutex` static as well, or it is not useful (it needs to be shared between all copies of the function that may run at the same time, but right now you have it a local variable).

You also need to expand the locked region until after the point where you actually *use* the random number generator. The easiest way is to use a scoped lock.

Finally, use `Threads::ThreadMutex` to stay compatible for non-C++11 compilers.